modules/account-lib/src/coin/sol/stakingBuilder.ts

```diff
@@ -0,0 +1,114 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import BigNumber from 'bignumber.js';
+import { TransactionType } from '../baseCoin';
+import { Transaction } from './transaction';
+import { TransactionBuilder } from './transactionBuilder';
+import { InstructionBuilderTypes } from './constants';
+
+import assert from 'assert';
+import { StakingDelegate, StakingInitialize } from './iface';
+import { isValidPublicKey } from './utils';
+import { BuildTransactionError } from '../baseCoin/errors';
+
+export class StakingBuilder extends TransactionBuilder {
+  protected _amount: string;
+  protected _controller: string;
+  protected _validator: string;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  protected get transactionType(): TransactionType {
+    return TransactionType.StakingActivate;
+  }
+
+  initBuilder(tx: Transaction): void {
+    super.initBuilder(tx);
+    for (const instruction of this._instructionsData) {
+      if (instruction.type === InstructionBuilderTypes.StakingInitialize) {
+        const initializeInstruction: StakingInitialize = instruction;
+        this.sender(initializeInstruction.params.fromAddress);
+        this.owner(initializeInstruction.params.stakeAddress);
+        this.amount(initializeInstruction.params.amount);
+      }
+      if (instruction.type === InstructionBuilderTypes.StakingDelegate) {
+        const delegateInstruction: StakingDelegate = instruction;
+        this.validator(delegateInstruction.params.validator);
+      }
+    }
+  }
+
+  /**
+   * The amount to stake expressed in Lamports, 1 SOL = 1_000_000_000
+   *
+   * @param {string} amount expressed in Lamports
+   * @returns {StakeBuilder} This staking builder.
+   *
+   */
+  amount(amount: string): this {
+    this.validateValue(new BigNumber(amount));
+    this._amount = amount;
+    return this;
+  }
+
+  /**
+   * The controller of the staking account.
+   *
+   * @param {BaseAddress} controller address of the stake account controller
+   * @returns {StakeBuilder} This staking builder.
+   *
+   * @see https://docs.solana.com/staking/stake-accounts#account-address
+   */
+  owner(controller: string): this {
```

Comment:
This method name is following dot staking implementation naming but this is kinda of weird here, will need help to choose a name.